src/modules/person/persistence/person.repository.ts

```diff
@@ -398,16 +405,21 @@ export class PersonRepository {
         return mapEntityToAggregate(personEntity);
     }
 
-    private hasChangedNames(personEntity: PersonEntity, person: Person<true>): boolean {
-        const oldVorname: string = personEntity.vorname.toLowerCase();
-        const oldFamilienname: string = personEntity.familienname.toLowerCase();
-        const newVorname: string = person.vorname.toLowerCase();
-        const newFamilienname: string = person.familienname.toLowerCase();
+    private hasUsernameChanged(
+        oldVorname: string,
+        oldFamilienname: string,
+        newVorname: string,
+        newFamilienname: string,
+    ): boolean {
+        const oldVornameLowerCase: string = oldVorname.toLowerCase();
+        const oldFamiliennameLowerCase: string = oldFamilienname.toLowerCase();
+        const newVornameLowerCase: string = newVorname.toLowerCase();
+        const newFamiliennameLowerCase: string = newFamilienname.toLowerCase();
 
         //only look for first letter, because username is firstname[0] + lastname
-        if (oldVorname[0] !== newVorname[0]) return true;
+        if (oldVornameLowerCase[0] !== newVornameLowerCase[0]) return true;
 
-        return oldFamilienname !== newFamilienname;
+        return oldFamiliennameLowerCase !== newFamiliennameLowerCase;
```

Comment:
This *might* return true for some rare cases, where the username would not actually change (for example, changing the first name `Ömer` to `Oemer` or the other way around, the resulting username would always start with `o`).
Consider not only lowercasing the names but also applying our normalization functions `toDIN91379SearchForm`

Reply:
Okay, does this apply for the PersonRenamedEvent also?